homeassistant/components/nina/config_flow.py

```diff
@@ -0,0 +1,138 @@
+    """Handle a config flow for NINA."""
+
+    VERSION: int = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        super().__init__()
+        self._all_region_codes_sorted: dict[str, str] = {}
+        self.regions: dict[str, dict[str, Any]] = {}
+
+        for name in CONST_REGIONS:
+            self.regions[name] = {}
+
+    async def async_step_user(
+        self: ConfigFlow,
+        user_input: dict[str, Any] | None = None,
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, Any] = {}
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        has_error: bool = False
+
+        if len(self._all_region_codes_sorted) == 0:
+            try:
+                nina: Nina = Nina()
+
+                self._all_region_codes_sorted = self.swap_key_value(
+                    await nina.getAllRegionalCodes()
+                )
+
+                self.split_regions()
+
+            except ApiError as err:
+                _LOGGER.warning("NINA setup error: %s", err)
+                errors["base"] = "cannot_connect"
+                has_error = True
+            except Exception as err:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception: %s", err)
+                errors["base"] = "unknown"
+                return self.async_abort(reason="unknown")
+
+        if user_input is not None and not has_error:
+            config: dict[str, Any] = user_input
+
+            config[CONF_REGIONS] = []
+
+            for group in CONST_REGIONS:
+                if group_input := user_input.get(group):
+                    config[CONF_REGIONS] += group_input
+
+            if len(config[CONF_REGIONS]) > 0:
+                tmp: dict[str, Any] = {}
+
+                for reg in config[CONF_REGIONS]:
+                    tmp[self._all_region_codes_sorted[reg]] = reg.split("_", 1)[0]
+
+                compact: dict[str, Any] = {}
+
+                for key, val in tmp.items():
+                    if val in compact:
+                        compact[val] = f"{compact[val]} + {key}"
+                        break
+                    compact[val] = key
+
+                config[CONF_REGIONS] = compact
+
+                return self.async_create_entry(title="NINA", data=config)
+
+            errors["base"] = "no_selection"
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    **{
+                        vol.Optional(region): cv.multi_select(self.regions[region])
+                        for region in CONST_REGIONS
+                    },
+                    vol.Required(CONF_MESSAGE_SLOTS, default=5): vol.All(
+                        int, vol.Range(min=1, max=20)
+                    ),
+                    vol.Required(CONF_FILTER_CORONA, default=True): cv.boolean,
```

Comment:
Why do we need this filter? Can we create all entities instead and let the user disable some that they don't want?

Reply:
I think it is a good idea to filter out the warnings about Corona directly and across the board, because they usually do not contain important information. I still wanted to give the user the possibility to disable this filter in case the information is of interest to him. To create all entities and then deactivate them has the disadvantage that when the corona warnings disappear the normal warning is not shown because the entity is disabled.

Reply:
Ok.